File: java/WEB-INF/classes/chat/PostMessage.java
package chat;

import java.io.*;
import java.util.*;
import javax.servlet.*;
import javax.servlet.http.*;
import chat.DB;
import org.json.simple.*;
import java.sql.*;
 
public class PostMessage extends HttpServlet {
	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException{
		

		// Opens the current users session
		HttpSession session = request.getSession(true);

		DB database = null;
		PrintWriter output = null;

		try{
			String newMessage = request.getParameter("msg");
			Integer userid = (Integer)session.getValue("userid");

			if(userid != null){
				database = new DB();

				Boolean result = database.postMessage(userid, newMessage);

				JSONObject obj = new JSONObject();
				obj.put("success", result);

				StringWriter out = new StringWriter();
				JSONValue.writeJSONString(obj, out);
				String jsonText = out.toString();

				response.setContentType("application/json;charset=UTF-8");
				output = response.getWriter();
				output.println(jsonText);
			}

		}catch(Exception e){
			output = response.getWriter();
			output.println("ERROR: " + e);
		}finally{
			try{
				database.close();
				output.close();
			}catch(Exception e){}
		}
	}
}

File: php/postMessage.php
<?php
	session_start();
	if( !isset($_SESSION['userid']) ){
		echo json_encode( array("success" => false) );
	}

	include('DB.php');
	include('Chat.php');
	$chat = new Chat(new DB());

	$message = isset($_POST['msg']) ? $_POST['msg'] : false;

	if($message){
		$message_success = $chat->postMessage($message);

		echo json_encode( array("success" => $message_success) );
	}else{
		echo json_encode( array("success" => false) );
	}

?>